Objective-C/CBLReplicator.mm

```diff
@@ -713,33 +607,13 @@ - (BOOL) active {
 
 - (void) setSuspended: (BOOL)suspended {
     CBL_LOCK(self) {
-        if (_state <= kCBLStateStopping) {
-            CBLLogInfo(Sync, @"%@: Ignore suspended = %d (state = %d, status = %d)",
-                       self, suspended, _state, _rawStatus.level);
-            return;
-        }
-        
-        BOOL changed = suspended != (_state == kCBLStateSuspended || _state == kCBLStateSuspending);
-        if (changed) {
-            CBLLogInfo(Sync, @"%@: Set suspended = %d (state = %d, status = %d)",
-                       self, suspended, _state, _rawStatus.level);
-            if (_state == kCBLStateSuspending) {
-                _cancelSuspending = !suspended;
-                return;
-            }
-            
-            if (suspended) {
-                _state = kCBLStateSuspending;
-                [self _stop];
-            } else
-                [self resetAndScheduleRetry];
-        }
+        c4repl_setSuspended(_repl, suspended);
```

Comment:
@jayahariv Have you looked if the logic to [1] ignore the suspend when the replicator is in the stop process or [2] to resume from suspending (e.g. switching fast b/w BG and FG) is still working?

At least we should create a ticket to verify after all other unit tests are passed.

Reply:
sure, will create a ticket to verify this

Reply:
https://issues.couchbase.com/browse/CBL-687